### PR TITLE
changed $should_cache default

### LINKED ISSUE #266: Cachify caches output from WordPress with HTTP-Status-Code 500, but shouldnt
We are using the nginx-hdd Version of cachify and triggered a problem.
We had a project which produced in frontend a critical error message, which was handled by WordPress  with his ""There has been a critical error on this website." output, delivered with a http-status-code 500.

But cachify saved the output and next time, nginx found the saved file and delivered it as expected with http-status-code 200.

So the problem is, that there shouldnt be a cache-file created if the http-status-code is not 200 in my opinion.
Perhaps you should add this check vor error-handling before you write the cache.

Thanks a lot

https://github.com/pluginkollektiv/cachify/blob/9ea110bfc39e677793884076b42b3a14f213da77/inc/class-cachify.php#L1395

### PR DESCRIPTION
**Pull request by karlkowald**
from default to statuscode matching 200

### CODE REVIEW

File: inc/class-cachify.php [hunk 1]
```diff
@@ -1392,7 +1392,7 @@ public static function set_cache( $data ) {
 		 * @param string $cache_hash    The cache hash.
 		 * @param int    $cache_expires Cache validity period.
 		 */
-		$should_cache = apply_filters( 'cachify_store_item', true, $data, self::$method, self::_cache_hash(), self::_cache_expires() );
+		$should_cache = apply_filters( 'cachify_store_item', http_response_code() === 200, $data, self::$method, self::_cache_hash(), self::_cache_expires() );
```
Comment (stklcode) [inline inc/class-cachify.php]: [`http_response_code`](https://www.php.net/manual/function.http-response-code.php) was introduced in PHP 5.4

`Cachify` is yet compatible with 5.2

If we merge this for the next point or major release (likely 2.4), I'd vote for raising the minimum PHP version so something more reasonable above 5.4 and ignore this issue for now.

**Comment by stklcode** [comment]
Besides the minor PHP version issue, code looks good to me. Thanks very much for this fix @karlkowald.

In general excluding 500s (which was the original intent) should be valid in almost any scenario.

Excluding errors like 404 is also a good idea in many cases. Caching them allows users/bots to fill up the cache requesting arbitrary, non-existing URLs. In the very same scene this wastes some processing time to not cache them.

I'm totally fine with a 200-only default here.

File: inc/class-cachify.php (hunk 1 above)
Comment (stklcode) [reply inc/class-cachify.php]: Seems we already have a vote for this in #259

So let's do it.

**Review by stklcode — APPROVED** [review]

File: inc/class-cachify.php (hunk 1 above)
Comment (pfefferle) [reply inc/class-cachify.php]: only one little thing: WordPress prefers yoda style checks

Comment (stklcode) [reply inc/class-cachify.php]: If we're talking about style, we should add some line breaks, too.

**Review by pfefferle — APPROVED** [review]
LGTM!

**Pull request merged**